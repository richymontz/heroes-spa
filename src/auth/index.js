// export * from './components';
// export * from './helpers';
export * from './pages';
export * from './context';
export * from './types';
// export * from './hooks';