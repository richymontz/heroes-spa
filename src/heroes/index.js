export * from './components';
// export * from './helpers';
export * from './pages';
export * from './routes';
// export * from './hooks';