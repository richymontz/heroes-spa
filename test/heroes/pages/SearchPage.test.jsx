import { fireEvent, render, screen } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"
import { SearchPage } from "../../../src/heroes"

const mockUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockUseNavigate,
}))


describe('<SearchPage/>', () => {
  test('should shows the page with default values', () => {
    const { container } = render(
      <MemoryRouter>
        <SearchPage></SearchPage>
      </MemoryRouter>
    )

    expect(container).toMatchSnapshot();
  })

  test('should show batman and query input should have batman', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=batman']}>
        <SearchPage></SearchPage>
      </MemoryRouter>
    )

    const input = screen.getByRole('textbox')
    expect(input.value).toBe('batman')


    const alert = screen.getByLabelText('alert-danger')
    expect(alert.style.display).toBe('none')

  })

  test('should show error if any hero found', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=someone']}>
        <SearchPage></SearchPage>
      </MemoryRouter>
    )

    const alert = screen.getByLabelText('alert-danger')
    expect(alert.style.display).toBe('');

    const input = screen.getByRole('textbox')
    expect(input.value).toBe('someone')
  })

  test('should navigates to new page', () => {
    render(
      <MemoryRouter initialEntries={['/search']}>
        <SearchPage></SearchPage>
      </MemoryRouter>
    )

    const input = screen.getByRole('textbox')

    fireEvent.change(input, { target: { name: input.name, value: 'batman' } })

    const form = screen.getByRole('form')
    fireEvent.submit(form)

    expect(mockUseNavigate).toHaveBeenCalledWith('?q=batman');
  })
})
