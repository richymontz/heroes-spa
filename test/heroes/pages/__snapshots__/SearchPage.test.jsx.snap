// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchPage/> should shows the page with default values 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-6"
    >
      <form
        aria-label="form"
        class="form-inline"
      >
        <div
          class="form-group mb-2"
        >
          <input
            autocomplete="off"
            class="form-control form-control-lg"
            name="searchText"
            placeholder="Tyoe hero name here..."
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group mx-sm-3 mb-2"
        >
          <button
            class="btn btn-primary btn-lg"
          >
             Search
          </button>
        </div>
      </form>
    </div>
    <div
      class="col-6"
    >
      <div
        aria-label="alert-danger"
        class="alert alert-danger"
        style="display: none;"
      >
        No Heroes found with 
        <b />
      </div>
    </div>
  </div>
</div>
`;
